webapp/channels/src/components/channel_view/index.ts

```diff
@@ -33,6 +34,8 @@ function mapStateToProps(state: GlobalState) {
     const enableOnboardingFlow = config.EnableOnboardingFlow === 'true';
     const enableWebSocketEventScope = config.FeatureFlagWebSocketEventScope === 'true';
 
+    const channelRoles = channel ? getMyChannelMembership(state, channel.id)?.roles || '' : '';
+    const channelRoleLoaded = channelRoles.split(' ').some((v) => Boolean(getRoles(state)[v]));
```

Comment:
Could you pull this out into a selector like `isDeactivatedChannel`? It doesn't need to be memoized, but it'd be nice if it was self-contained since this logic is fairly complex despite being only 2 lines long